File: plugins/script_loader/source/script_api/v1/bookmarks.cpp
#include <script_api.hpp>

#include <hex/api/imhex_api.hpp>

#define VERSION V1

SCRIPT_API(void createBookmark, u64 address, u64 size, u32 color, const char *name, const char *description) {
    hex::ImHexApi::Bookmarks::add(address, size, name, description, color);
}

File: plugins/script_loader/templates/CSharp/ImHexLibrary/Bookmarks.cs
using System.Drawing;
using System.Runtime.InteropServices;
using System.Text;

namespace ImHex
{
    public partial class Bookmarks
    {
        [LibraryImport("ImHex")]
        private static partial void createBookmarkV1(UInt64 address, UInt64 size, UInt32 color, byte[] name, byte[] description);

        public static void CreateBookmark(long address, long size, Color color, string name = "", string description = "")
        {
            createBookmarkV1((UInt64)address, (UInt64)size, (UInt32)(0xA0 << 24 | color.B << 16 | color.G << 8 | color.R), Encoding.UTF8.GetBytes(name), Encoding.UTF8.GetBytes(description));
        }

    }
}
